### ISSUE TITLE
Why a seperate etcd server?

### ISSUE THREAD

**Issue by nathansamson** [hermanjunge/k8s-do-tf#6]
This seems a bit like a waste of resources...

Also; does this mean this one etcd server is SPOF?

**Comment by ghost** [comment]
Hi.

> This seems a bit like a waste of resources...

Yes, it seems like a waste of resources. I agree. Let's say that we were following to the letter the guide of setting a k8s cluster [provided by coreOS](https://coreos.com/kubernetes/docs/latest/getting-started.html).

However, if we don't want to spend two servers, we can easily setup `etcd` and `master` in the same machine, without losing anything. _BUT_ we will find that we have reduced both functions into a single SPOF!

> Also; does this mean this one etcd server is SPOF?

Yes, it is a Single point of failure. The alternative is to setup `etcd` as a cluster. There are a number of guides over there that setup 3 servers to comply that function.

**TLDR**: It all depends on how many servers you want to invest for your cluster. You can even have kubernetes in one single server. There is no such thing as the _right_ way to kubernetes.
